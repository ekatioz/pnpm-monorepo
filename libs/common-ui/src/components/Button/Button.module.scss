.button {
  background-color: #333;
  border: 2px solid black;
  color: white;
  padding: 0.5rem 1rem;
  border-radius: 8px;
  cursor: pointer;
}
