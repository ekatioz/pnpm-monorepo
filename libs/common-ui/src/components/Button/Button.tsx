import styles from "./Button.module.scss";

type ButtonProps = {
  caption: string;
};

export function Button({ caption }: ButtonProps) {
  return <button className={styles.button}>{caption}</button>;
}
