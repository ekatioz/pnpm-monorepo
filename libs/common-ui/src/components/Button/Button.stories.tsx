import type { Meta, StoryObj } from "@storybook/react";
import { useEffect } from "react";
import { Button } from "./Button";

export default { component: Button } satisfies Meta<typeof Button>;

export const Primary: StoryObj<typeof Button> = {
  render: (args) => {
    useEffect(() => {
      console.log("mount");
      return () => {
        console.log("unmount");
      };
    }, []);
    return <Button {...args} />;
  },
  args: {
    caption: "Hallo",
  },
};
