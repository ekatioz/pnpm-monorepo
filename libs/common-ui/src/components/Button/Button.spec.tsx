import { render } from "@testing-library/react";
import { Button } from "./Button";

it("renders", () => {
  const { container } = render(<Button caption="Click" />);
  expect(container).toMatchSnapshot();
});
