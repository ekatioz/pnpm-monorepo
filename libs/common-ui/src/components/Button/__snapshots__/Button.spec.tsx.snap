// Vitest Snapshot v1

exports[`renders 1`] = `
<div>
  <button
    class="_button_f8f296"
  >
    Click
  </button>
</div>
`;
